pkg/apis/testing/v1alpha1/testsuite_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TestSuiteSpec defines the desired state of TestSuite
+type TestSuiteSpec struct {
+	// How many tests we want to execute at the same time.
+	// Depends on cluster size and it's load
+	ConcurrencyLevel int64 `json:"concurrencyLevel,omitempty"`
+	// You can specify test names explicitly.
+	TestNamesSelector []NamespacedTest `json:"testNamesSelector,omitempty"`
+	// Will run every test that depends of any of the component listed here
+	ComponentsSelector []string `json:"componentsSelector,omitempty"`
+	// Running all tests from suite cannot take more time that specified here
+	SuiteTimeout *metav1.Duration `json:"suiteTimeout,omitempty"`
+	// If specific TestDefinition does not define timeout, use this one
+	DefaultTestTimeout *metav1.Duration `json:"defaultTestTimeout,omitempty"`
+	// Should I repeat every test? Default value will be 1
+	Repeat int64 `json:"repeat,omitempty"`
+	// In case of a failed test, how many times it will be retried.
+	// If test failed and on retry it succeeded, Test Suite should be marked as a succeeded.
+	MaxRetries int64 `json:"maxRetries,omitempty"`
+}
+
+type NamespacedTest struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+// TestSuiteStatus defines the observed state of TestSuite
+type TestSuiteStatus struct {
+	StartTime      *metav1.Time         `json:"startTime,omitempty"`
+	CompletionTime *metav1.Time         `json:"completionTime,omitempty"`
+	Conditions     []TestSuiteCondition `json:"conditions,omitempty"`
+	Results        []TestResult         `json:"results,omitempty"`
+}
+
+type TestSuiteConditionType string
+type Status string
+type TestResultConditionType string
+
+const (
+	StatusTrue    Status = "true"
```

Comment:
minor: value should start with capital letter: `True, False, or Unknown`

source: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties